### ISSUE TITLE
ivhd option doesn't accommodate empty X list

### ISSUE THREAD

**Issue by markeschaffer** [aahrens1/ddml#16]
SSIA, as they used to say.

**Comment by aahrens1** [comment]
It's sort of supported:

```
*** load & prepare data
use https://github.com/aahrens1/ddml/raw/master/data/BLP.dta, clear
global Y share
global D price
global Z sum*

*** select model
set seed 42
ddml init ivhd

*** add learners
ddml E[Y|X]: reg $Y  
ddml E[D|Z,X], learner(Dhat_pystacked): ///
	pystacked $D $Z, type(reg) method(lassocv)
ddml E[D|X], learner(Dhat_pystacked) vname($D): reg {D} 

*** cross-fitting and estimation
ddml crossfit
ddml estimate
```

But we probably want to provide a more user-friendly syntax. How about this?

```
ddml init ivhd, nocontrols
ddml E[D|Z]: pystacked $D $Z, type(reg) method(lassocv)
ddml crossfit
ddml estimate
```

**Comment by markeschaffer** [comment]
There's also the constant though. That's part of "X". The first syntax will partial out the constant, unless regress with nocons is the learner [sic].

Maybe the second syntax could have a nocons option.  It also needs to specify the dep var somewhere.

**Comment by aahrens1** [comment]
True. -- But do we want to support `nocons`? I cannot imagine a scenario where that's a good idea.

**Comment by markeschaffer** [comment]
Pretty unlikely `nocons` would be useful. But that can be done using your syntax #1 anyway.

Maybe it's not worth adding the option just now, and instead just explain how to do it using syntax #1. It's sort-of instructive as an example.

**Comment by aahrens1** [comment]
I have also added support for no controls to `qddml`. The long version shows what `qddml` is doing internally. 

```
use https://github.com/aahrens1/ddml/raw/master/data/BLP.dta, clear
global Y share
global D price
global Z sum*

*** long version
set seed 42
ddml init ivhd, kfolds(2)
ddml E[Y|X]: reg $Y  
ddml E[D|Z,X], learner(Dhat_pystacked): ///
	pystacked $D $Z, type(reg)
ddml E[D|X], learner(Dhat_pystacked) vname($D): reg {D} 
ddml crossfit
ddml estimate

*** short version
set seed 42
qddml $Y ($D = $Z), model(ivhd) kfolds(2) cmd(pystacked) 
```
